src/sorcha/ephemeris/simulation_driver.py

```diff
@@ -70,98 +74,59 @@ def create_ephemeris(orbits_df, pointings_df, args, configs):
     for _, pointing in pointings_df.iterrows():
         mjd_tai = float(pointing["observationStartMJD"])
         ra, dec = float(pointing["fieldRA"]), float(pointing["fieldDec"])
-        visit_vec = ra_dec2vec(ra, dec)
-        pixels = get_hp_neighbors(ra, dec, ang_fov + buffer, nside=nside, nested=True)
-        jd_tdb = mjd_tai_to_epoch(mjd_tai)
-        et = (jd_tdb - spice.j2000()) * 24 * 60 * 60
-        utc_str = spice.et2utc(et, "J", 7)
-
-        if False:
-            r_obs = observatories.barycentricObservatory(et, obsCode) / AU_KM
-        else:
-            r_obs, v_obs = barycentricObservatoryRates(
-                et, obsCode, observatories=observatories
-            )  # <--- Need this routine
-            r_obs /= AU_KM  # convert to au
-            v_obs *= (24 * 60 * 60) / AU_KM  # convert to au/day
-            sun = ephem.get_particle("Sun", jd_tdb - ephem.jd_ref)  # This can be pre-calculated
-            r_sun = np.array((sun.x, sun.y, sun.z))
-            v_sun = np.array((sun.vx, sun.vy, sun.vz))
 
         # If the observation time is too far from the
         # time of the last set of ballpark sky position,
         # compute a new set
         while (
-            abs(jd_tdb - t_picket) > 0.5 * picket_interval or first == 1
+            abs(pointing["jd_tdb"] - t_picket) > 0.5 * picket_interval or first == 1
```

Comment:
Most of the changes from here down are simply making use of the precomputed values stored in the `pointing` dataframe row.